File: src/main/java/no/nav/fo/veilarbregistrering/registrering/bruker/OrdinaerBrukerRegistreringMetrikker.java
package no.nav.fo.veilarbregistrering.registrering.bruker;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import no.nav.fo.veilarbregistrering.metrics.Events;
import no.nav.fo.veilarbregistrering.metrics.Metric;
import no.nav.fo.veilarbregistrering.metrics.MetricsService;

class OrdinaerBrukerRegistreringMetrikker {
    static void rapporterInvalidRegistrering(MetricsService metricsService, OrdinaerBrukerRegistrering ordinaerBrukerRegistrering) {
        metricsService.reportFields(Events.INVALID_REGISTRERING_EVENT,
                Metric.of("registrering", toJson(ordinaerBrukerRegistrering.getBesvarelse())),
                Metric.of("stilling", toJson(ordinaerBrukerRegistrering.getSisteStilling())));
    }

    private static String toJson(Object obj) {
        String json = "";
        try {
            json = (new ObjectMapper()).writeValueAsString(obj);
        } catch (JsonProcessingException ignored) {
        }
        return json;
    }
}

File: src/main/java/no/nav/fo/veilarbregistrering/registrering/bruker/OrdinaerBrukerRegistreringMetrikker.kt
package no.nav.fo.veilarbregistrering.registrering.bruker

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import no.nav.fo.veilarbregistrering.metrics.Events
import no.nav.fo.veilarbregistrering.metrics.Metric.Companion.of
import no.nav.fo.veilarbregistrering.metrics.InfluxMetricsService

internal object OrdinaerBrukerRegistreringMetrikker {
    @JvmStatic
    fun rapporterInvalidRegistrering(
            influxMetricsService: InfluxMetricsService,
            ordinaerBrukerRegistrering: OrdinaerBrukerRegistrering
    ) {
        influxMetricsService.reportFields(
            Events.INVALID_REGISTRERING_EVENT,
            of("registrering", toJson(ordinaerBrukerRegistrering.getBesvarelse())),
            of("stilling", toJson(ordinaerBrukerRegistrering.getSisteStilling()))
        )
    }

    private fun toJson(obj: Any): String =
        try {
            ObjectMapper().writeValueAsString(obj)
        } catch (ignored: JsonProcessingException) {
            ""
        }
}